tsdb/series_segment.go

```diff
@@ -120,14 +126,16 @@ func (s *SeriesSegment) Path() string { return s.path }
 // InitForWrite initializes a write handle for the segment.
 // This is only used for the last segment in the series file.
 func (s *SeriesSegment) InitForWrite() (err error) {
-	// Only calculcate segment data size if writing.
-	for s.size = uint32(SeriesSegmentHeaderSize); s.size < uint32(len(s.data)); {
-		flag, _, _, sz := ReadSeriesEntry(s.data[s.size:])
+	// Only recalculate segment data size if writing.
+	var size uint32
+	for size = uint32(SeriesSegmentHeaderSize); size < s.size; {
+		flag, _, _, sz := ReadSeriesEntry(s.data[size:])
```

Comment:
`ReadSeriesEntry` could read beyond the end of the file into the rounded-up page or into SIGBUS land here. Even if the slice is bounded to the physical size of the file (`s.data[size:s.size]`) you could get an out of bounds panic from `ReadSeriesEntry` because it does not perform bounds checking.

Reply:
Good catch. Fix coming.

Reply:
That panic could only happen in a corrupt file, I think, right?

Reply:
I now think that this can't happen.  Every place we call `ReadSeriesEntry` we know we have at least one byte in the array. 

https://github.com/influxdata/influxdb/blob/e82ffebc0abaae5bf29c2f5c366fb42a4b1b3adc/tsdb/series_segment.go#L254
https://github.com/influxdata/influxdb/blob/e82ffebc0abaae5bf29c2f5c366fb42a4b1b3adc/tsdb/series_segment.go#L131
https://github.com/influxdata/influxdb/blob/e82ffebc0abaae5bf29c2f5c366fb42a4b1b3adc/cmd/influx_inspect/verify/seriesfile/verify.go#L220

We will crash with a bounds check in `binary.BigEndian.Uint64` if we don't have enough bytes, or in `ReadSeriesKey` but I don't see how to avoid that, because we don't know how many bytes we need to unpack to get the uint64 in `binary.Uvarint`, but I think that's an inevitable fact of a corrupt file.

Reply:
The test is about partial writes, so I think it is a deliberate attempt to have a too-short file. But, it doesn't generate a SIGBUS, so I'm not sure if it is walking off the end of the file, or whether I am putting the wrong limit on the slice size.